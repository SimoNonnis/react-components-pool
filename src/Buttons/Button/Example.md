It renders a button with required props

```jsx
const onClick = () => true;
<Button onClick={onClick}>Button</Button>;
```

It renders a button with props size "small" and "disabled" true

```jsx
const onClick = () => true;
<Button onClick={onClick} size="small" disabled>
  Button
</Button>;
```

It renders a button with props "disabled" true, "loading" true and size="small"

```jsx
const onClick = () => true;
<Button onClick={onClick} disabled loading size="small">
  Button
</Button>;
```

It renders a button with props "disabled" true and "loading" true

```jsx
const onClick = () => true;
<Button onClick={onClick} disabled loading>
  Button
</Button>;
```

It renders a button with props "disabled" true, "loading" true and size="large"

```jsx
const onClick = () => true;
<Button onClick={onClick} disabled loading size="large">
  Button
</Button>;
```

It renders a button with props styleType="danger"

```jsx
const onClick = () => true;
<Button onClick={onClick} styleType="danger">
  Button
</Button>;
```

It renders a button with props type="danger"

```jsx
const onClick = () => true;
<Button onClick={onClick} type="danger">
  Button
</Button>;
```

It renders a button with props type="secondary"

```jsx
const onClick = () => true;
<Button onClick={onClick} type="secondary">
  Button
</Button>;
```

It renders a button with props size "large" and type="highlight"

```jsx
const onClick = () => true;
<Button onClick={onClick} size="large" type="highlight">
  Button
</Button>;
```

It renders a button with rounded borders

```jsx
const onClick = () => true;
<Button onClick={onClick} rounded={50}>
  Button
</Button>;
```

It renders a button with props type="highlight" and fullWidth 100%

```jsx
const onClick = () => true;
<Button onClick={onClick} type="highlight" fullWidth>
  Button
</Button>;
```
It renders a button with props styleType="link" type="button"

```jsx
const onClick = () => console.log('i look like a link but I am a button!!');
<Button onClick={onClick} styleType="link" type="button">
  Button
</Button>;
```

It renders a button with props type="link"

```jsx
const onClick = () => console.log('i look like a link but I am a button!!');
<Button onClick={onClick} type="link">
  Button
</Button>;
```
