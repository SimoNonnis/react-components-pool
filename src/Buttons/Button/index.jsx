import React from 'react';
import PropTypes from 'prop-types';

import Loader from '../_common/Loader';

import { Btn } from '../style';

const propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  disabled: PropTypes.bool,
  loading: PropTypes.bool,
  fullWidth: PropTypes.bool,
  onClick: PropTypes.func.isRequired,
  rounded: PropTypes.number,
  size: PropTypes.oneOf(['small', 'large', '']),
  styleType: PropTypes.oneOf(['secondary', 'highlight', 'outline', 'white', 'danger', 'link', '']),
  type: PropTypes.oneOf([
    'secondary',
    'highlight',
    'outline',
    'white',
    'danger',
    'link',
    'button',
    'submit',
  ]),
};

const defaultProps = {
  className: '',
  disabled: false,
  loading: false,
  fullWidth: false,
  rounded: 3,
  size: '',
  styleType: '',
  type: 'submit',
};

function Button({
  children,
  className,
  disabled,
  loading,
  fullWidth,
  onClick,
  rounded,
  size,
  styleType,
  type,
}) {
  const buttonChildren = loading ? <Loader size={size} /> : children;
  return (
    <Btn
      className={className}
      disabled={disabled}
      fullWidth={fullWidth}
      onClick={onClick}
      rounded={rounded}
      size={size}
      styleType={styleType}
      type={type}
    >
      {buttonChildren}
    </Btn>
  );
}

Button.propTypes = propTypes;
Button.defaultProps = defaultProps;

export default Button;
