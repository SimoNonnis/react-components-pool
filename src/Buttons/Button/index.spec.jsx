/* eslint-env jest */
import React from 'react';

import 'jest-styled-components';

import Button from './';

describe('Test <Button />', () => {
  const mockOnClick = jest.fn();
  it('should render Button correctly with required props', () => {
    const wrapper = shallow(<Button onClick={mockOnClick}>Click me</Button>);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render Button correctly with props size "small" and "disabled" true', () => {
    const wrapper = shallow(
      <Button onClick={mockOnClick} size="small" disabled>
        Click me
      </Button>
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render Button correctly with props type="button"', () => {
    const wrapper = shallow(
      <Button onClick={mockOnClick} type="button">
        Click me
      </Button>
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render Button correctly with props type="submit"', () => {
    const wrapper = shallow(
      <Button onClick={mockOnClick} type="submit">
        Click me
      </Button>
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render Button correctly with props styleType="secondary and type="button"', () => {
    const wrapper = shallow(
      <Button onClick={mockOnClick} styleType="secondary" type="button">
        Click me
      </Button>
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render Button correctly with props type="secondary', () => {
    const wrapper = shallow(
      <Button onClick={mockOnClick} type="secondary">
        Click me
      </Button>
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render correctly a Button of type "highlight" with 100% width', () => {
    const wrapper = shallow(
      <Button onClick={mockOnClick} type="outline" fullWidth>
        Click me
      </Button>
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render Button correctly with props size "small" and "disabled" true', () => {
    const wrapper = shallow(
      <Button onClick={mockOnClick} size="small" disabled>
        Click me
      </Button>
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render Button correctly with props size "large" and type "highlight"', () => {
    const wrapper = shallow(
      <Button onClick={mockOnClick} size="large" type="highlight">
        Click me
      </Button>
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render Button correctly with custom rounded corners', () => {
    const wrapper = shallow(
      <Button onClick={mockOnClick} rounded={50}>
        Click me
      </Button>
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render correctly a Button of type "white"', () => {
    const wrapper = shallow(
      <Button onClick={mockOnClick} type="white">
        Click me
      </Button>
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render correctly a Button with loading true', () => {
    const wrapper = shallow(
      <Button onClick={mockOnClick} loading>
        Click me
      </Button>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
