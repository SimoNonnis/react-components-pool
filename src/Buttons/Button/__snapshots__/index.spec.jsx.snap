// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Button /> should render Button correctly with custom rounded corners 1`] = `
<Btn
  className=""
  disabled={false}
  fullWidth={false}
  onClick={[MockFunction]}
  rounded={50}
  size=""
  styleType=""
  type="submit"
>
  Click me
</Btn>
`;

exports[`Test <Button /> should render Button correctly with props size "large" and type "highlight" 1`] = `
<Btn
  className=""
  disabled={false}
  fullWidth={false}
  onClick={[MockFunction]}
  rounded={3}
  size="large"
  styleType=""
  type="highlight"
>
  Click me
</Btn>
`;

exports[`Test <Button /> should render Button correctly with props size "small" and "disabled" true 1`] = `
<Btn
  className=""
  disabled={true}
  fullWidth={false}
  onClick={[MockFunction]}
  rounded={3}
  size="small"
  styleType=""
  type="submit"
>
  Click me
</Btn>
`;

exports[`Test <Button /> should render Button correctly with props size "small" and "disabled" true 2`] = `
<Btn
  className=""
  disabled={true}
  fullWidth={false}
  onClick={[MockFunction]}
  rounded={3}
  size="small"
  styleType=""
  type="submit"
>
  Click me
</Btn>
`;

exports[`Test <Button /> should render Button correctly with props styleType="secondary and type="button" 1`] = `
<Btn
  className=""
  disabled={false}
  fullWidth={false}
  onClick={[MockFunction]}
  rounded={3}
  size=""
  styleType="secondary"
  type="button"
>
  Click me
</Btn>
`;

exports[`Test <Button /> should render Button correctly with props type="button" 1`] = `
<Btn
  className=""
  disabled={false}
  fullWidth={false}
  onClick={[MockFunction]}
  rounded={3}
  size=""
  styleType=""
  type="button"
>
  Click me
</Btn>
`;

exports[`Test <Button /> should render Button correctly with props type="secondary 1`] = `
<Btn
  className=""
  disabled={false}
  fullWidth={false}
  onClick={[MockFunction]}
  rounded={3}
  size=""
  styleType=""
  type="secondary"
>
  Click me
</Btn>
`;

exports[`Test <Button /> should render Button correctly with props type="submit" 1`] = `
<Btn
  className=""
  disabled={false}
  fullWidth={false}
  onClick={[MockFunction]}
  rounded={3}
  size=""
  styleType=""
  type="submit"
>
  Click me
</Btn>
`;

exports[`Test <Button /> should render Button correctly with required props 1`] = `
<Btn
  className=""
  disabled={false}
  fullWidth={false}
  onClick={[MockFunction]}
  rounded={3}
  size=""
  styleType=""
  type="submit"
>
  Click me
</Btn>
`;

exports[`Test <Button /> should render correctly a Button of type "highlight" with 100% width 1`] = `
<Btn
  className=""
  disabled={false}
  fullWidth={true}
  onClick={[MockFunction]}
  rounded={3}
  size=""
  styleType=""
  type="outline"
>
  Click me
</Btn>
`;

exports[`Test <Button /> should render correctly a Button of type "white" 1`] = `
<Btn
  className=""
  disabled={false}
  fullWidth={false}
  onClick={[MockFunction]}
  rounded={3}
  size=""
  styleType=""
  type="white"
>
  Click me
</Btn>
`;

exports[`Test <Button /> should render correctly a Button with loading true 1`] = `
<Btn
  className=""
  disabled={false}
  fullWidth={false}
  onClick={[MockFunction]}
  rounded={3}
  size=""
  styleType=""
  type="submit"
>
  <Loader
    size=""
  />
</Btn>
`;
